src/util/psbt/mod.rs

```diff
@@ -210,8 +214,221 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    /// Signs the input at `input_index` with private key `sk`.
+    pub fn sign<C: Signing>(&mut self, sk: &PrivateKey, input_index: usize, secp: &Secp256k1<C>) -> Result<(), SighashError> {
+        self.check_index_is_within_bounds(input_index)?;
+
+        let mut cache = SighashCache::new(&self.unsigned_tx);
+        let (msg, sighash_ty) = self.sighash(input_index, &mut cache, None)?;
+
+        let sig = secp.sign_ecdsa(&msg, &sk.inner);
+
+        let mut final_signature = Vec::with_capacity(75);
+        final_signature.extend_from_slice(&sig.serialize_der());
+        final_signature.push(sighash_ty.to_u32() as u8);
+
+        let pk = sk.public_key(secp);
+        self.inputs[input_index]
+            .partial_sigs
+            .insert(pk, EcdsaSig::from_slice(&final_signature)?);
+
+        Ok(())
+
```

Comment:
nit: think we can get rid of this extra line

Reply:
Woops, I looked at line 235 when I read your comment. 237 is definitely a typo by me, thanks.